### PR TITLE
EPMRPP-78271 || add multipleAutocomplete component

### PR DESCRIPTION
**Pull request by chivekrodis**

### CODE REVIEW

File: app/src/componentLibrary/autocompletes/multipleAutocomplete/multipleDownshift.jsx [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+ * limitations under the License.
+ */
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import Downshift from 'downshift';
+import isEqual from 'fast-deep-equal';
+import { isArray } from 'c3/src/util';
+
+export class MultipleDownshift extends Component {
```
Comment (tr1ble) [inline app/src/componentLibrary/autocompletes/multipleAutocomplete/multipleDownshift.jsx:23]: Why is not a functional component ?

**Review by tr1ble — APPROVED** [review]

File: app/src/componentLibrary/autocompletes/multipleAutocomplete/multipleDownshift.jsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+  const handleSelection = (selectedItem, downshift) => {
+    if (!selectedItem) return;
+    if (selectedItems.some((item) => isEqual(item, selectedItem))) {
+      removeItem(selectedItem, downshift);
+    } else {
+      addSelectedItem(selectedItem, downshift);
+    }
+  };
+  const getStateAndHelpers = (downshift) => {
+    return {
```
Comment (Vadim73i) [inline app/src/componentLibrary/autocompletes/multipleAutocomplete/multipleDownshift.jsx:85]: FYI: simplify return to ()

File: app/src/componentLibrary/autocompletes/multipleAutocomplete/selectedItems/selectedItems.jsx [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    }
+  };
+  const onBlurHandler = () => {
+    setEditMode(false);
+    setValue('');
+  };
+
+  /* eslint-disable jsx-a11y/no-autofocus */
+  return (
+    <>
```
Comment (Vadim73i) [inline app/src/componentLibrary/autocompletes/multipleAutocomplete/selectedItems/selectedItems.jsx:64]: FYI: Is fragment really needed here?

File: app/src/pages/inside/projectSettingsPageContainer/content/notifications/modals/addEditNotificationModal/launchNamesContainer/launchNamesContainer.jsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  const activeProject = useSelector(projectIdSelector);
+  const [showMessage, setShowMessage] = useState(false);
+
+  const handleSystemMessage = (items, storedItems) => {
+    if (items.length === Object.keys(storedItems).length) {
+      setShowMessage(false);
+    } else {
+      setShowMessage(true);
+    }
+  };
```
Comment (Vadim73i) [inline app/src/pages/inside/projectSettingsPageContainer/content/notifications/modals/addEditNotificationModal/launchNamesContainer/launchNamesContainer.jsx:61]: setShowMessage(items.length !== Object.keys(storedItems).length)

**Review by Vadim73i — CHANGES_REQUESTED** [review]

File: app/src/componentLibrary/autocompletes/multipleAutocomplete/selectedItems/selectedItems.jsx (hunk 3 above)
Comment (chivekrodis) [reply app/src/componentLibrary/autocompletes/multipleAutocomplete/selectedItems/selectedItems.jsx:64]: removed

File: app/src/componentLibrary/autocompletes/multipleAutocomplete/multipleAutocomplete.jsx [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+          inputValue,
+          highlightedIndex,
+          removeItem,
+          editItem,
+          openMenu,
+          selectItem,
+          clearSelection,
+          storedItemsMap,
+        }) => (
+          <div>
```
[the remark this answers is not in the document]
Comment (chivekrodis) [reply app/src/componentLibrary/autocompletes/multipleAutocomplete/multipleAutocomplete.jsx:114]: It is downshift requirement as children should contains only one child

File: app/src/common/utils/validation/validate.js [hunk 6]
```diff
@@ -104,7 +104,17 @@ export const healthCheckWidgetPassingRate = composeValidators([isNotEmpty, range
 export const flakyWidgetNumberOfLaunches = composeValidators([isNotEmpty, range(2, 600)]);
 export const launchesWidgetContentFields = composeValidators([isNotEmptyArray, minLength(4)]);
 export const mostFailedWidgetNumberOfLaunches = composeValidators([isNotEmpty, range(2, 600)]);
-export const createNotificationRecipientsValidator = (informOwner) => (value) =>
-  isNotEmptyArray(value) || informOwner;
+export const createNotificationRecipientsValidator = (informOwner) => (value = []) => {
+  if (!informOwner && !value.length) return false;
+  if (informOwner && !value.length) {
+    return true;
+  }
```
[the remark this answers is not in the document]
Comment (chivekrodis) [reply app/src/common/utils/validation/validate.js:111]: I have using built in config. Can you provide some more info

File: app/src/common/utils/validation/validate.js [hunk 7]
```diff
@@ -104,7 +104,17 @@ export const healthCheckWidgetPassingRate = composeValidators([isNotEmpty, range
...
+  const re = /[.@]/;
```
[the remark this answers is not in the document]
Comment (chivekrodis) [reply app/src/common/utils/validation/validate.js:112]: re is a regular expression
regex return a new function and I pass it as a function to some method
I have refactored it lil bit

**Review by Vadim73i — APPROVED** [review]

**Pull request merged**